integration-auth/src/main/java/org/symphonyoss/integration/authentication/jwt/JwtAuthentication.java

```diff
@@ -163,4 +202,56 @@ public boolean isValidTokenPair(String configurationId, String applicationToken,
     }
     return symphonyToken.equals(bothTokens.getSymphonyToken());
   }
+
+  /**
+   * Validate if the sent JWT is valid by checking its signer and decodification and then return it.
+   * @param jwt Json Web Token containing the user/app authentication data.
+   * @return JwtPayload parsed;
+   */
+  public JwtPayload parseJwtPayload(String jwt) {
+    PublicKey rsaVerifier = getPodPublicSignatureVerifier();
+    Jws<Claims> jws = null;
+    try {
+      jws = Jwts.parser().setSigningKey(rsaVerifier).parseClaimsJws(jwt);
+    } catch (ExpiredJwtException e) {
+      Date expiration = e.getClaims().getExpiration();
+      if (expiration.before(new Date())) {
```

Comment:
If Jwts always throws an exception when JWT is expired, why do we need to check the expiration date? Shouldn't the application request another JWT token to the cloud?